web_server/entity/xml_parser.py

```diff
@@ -39,7 +39,21 @@ def get_test_names(self):
         """
         :return: a list of the names of the test cases
         """
-        return [test["name"] for test in self.obj.robot.suite.suite.suite.test]
+        test_names = []
+        if self.obj is not None:
+            for suite1 in self.obj.robot.suite.suite:
+                try:
```

Comment:
Again, this code is not easy to read.  The best is to move the try except block to an initialization method as follows:

    def __init__(self, object_from_xml):
        """
        Parser for Robot XML object
        :param object_from_xml: Python object from a Robot XML file
        """
        self.obj = object_from_xml
        self.set_python_object()
        self.vtt_entry = f'<count>\n<start-time> --> <end-time>\n<keyword>\n\n'

    def set_python_object(self):
        assert self.obj is not None

        try:
            len(self.obj.robot.suite.suite.suite.test)
            self.obj = self.obj.robot.suite.suite.suite.test
        except AttributeError:
            self.obj = self.obj.robot.suite.suite.test

The change all the occurrences of self.obj.robot.suite.suite.suite.test for self.obj

Reply:
Thank you, I think this definitely looks more clear. We could use this for single_suite xml file, for multiple suites xml file, to get the test names, we need to loop through each suite in self.obj.robot.suite.suite, and loop through each test case in each self.obj.robot.suite.suite.suite, so we might still need to use try except for multiple suites xml in the other functions